### PR TITLE
WT-10300 Additional speedups for s_fast.

### PR DESCRIPTION
**Pull request by luke-pearson**
I've made a number of changes here:
- I created a new s_evergreen_validate script, I discussed merging this with s_evergreen but it doesn't come out nicely. If people would like me to still try I can.
- I only run s_evergreen_validate if git says the files are modified in fast mode
- I added a fast argument to s_function which reduced its runtime from 30 seconds to about .5 seconds (modulo git changes)
- I fixed a typo in that script that meant it wasn't running on a number of files
- Fixed the subsequent issue it found
- Removed a spurious echo

### CODE REVIEW

**Comment by luke-pearson** [comment]
After this change ./s_fast is running in about 10 seconds for me, I would like to reduce that to 1 second but I will move further fixes into another ticket.

File: dist/s_evergreen_validate [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+t=__wt.$$
+trap 'rm -f $t' 0 1 2 3 13 15
+
+fast=1
+case "$1" in
+    -F) # Run fast.
+        fast=0
+        shift;;
+    *)
+        if test $# -gt 1 ; then
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply dist/s_evergreen_validate:12]: That's fine.

File: dist/s_all [hunk 2]
```diff
@@ -144,7 +134,7 @@ for f in `find . -name ${t_pfx}\*`; do
         LOCAL_NAME=`basename $f`
         # Find original command and trim redirect garbage
         FAILED_CMD=`echo "$COMMANDS" | grep $LOCAL_NAME | \
-            sed -e 's/ >.*//' -e 's/.* //'`
+            sed -e 's/ >.*//' -e 's/2>&1 //'`
```
Comment (lukech) [inline dist/s_all:137]: Thanks for fixing it.

File: dist/s_evergreen_validate [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            echo "Usage: $0 [-F]"
+            echo "-F only run validation commands if evergreen yml files have been modified"
+        fi
+        break;;
+esac
+
+if [ $(command -v evergreen) ] && [ -f ~/.evergreen.yml ]; then
+    if test $fast -eq 0 ; then
+        # Check the evergreen.yml files for modifications.
+        search=`git diff --name-only $(git merge-base --fork-point develop) | grep -E 'evergreen.*\.yml$'`
```
Comment (lukech) [inline dist/s_evergreen_validate:22]: It seems there's an assumption the git command is run for a change against the `develop` branch. Is it the only branch the fast option is targeted for?

Comment (luke-pearson) [reply dist/s_evergreen_validate:22]: Correct

File: dist/s_evergreen_validate [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    exit1=$?
+    evergreen validate -p wiredtiger test/evergreen_develop.yml >> dist/$t 2>&1
+    exit2=$?
+    cd dist
+fi
+
+if test $exit1 -eq 1 -o $exit2 -eq 1 ; then
+    echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+    echo "$0 failed with output:"
+    cat $t
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply dist/s_evergreen_validate:39]: Yes you can tell fairly obviously:
```
    ./s_evergreen_validate failed with output:
    A new version is available. Run 'evergreen get-update --install' to download and install it.
    test/evergreen.yml is valid
    A new version is available. Run 'evergreen get-update --install' to download and install it.
    ERROR: unmarshalling project config from YAML: invalid configuration file: yaml: line 29: did not find expected key
    ERROR: load project error(s): error unmarshalling yaml strict: yaml: line 29: did not find expected key
    test/evergreen_develop.yml is an invalid configuration
```

[the remark this answers is not in the document]
Comment (luke-pearson) [reply dist/s_evergreen_validate:39]: I have added some information for differentiation:
```
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
./s_evergreen_validate failed with output:
Validating evergreen.yml
A new version is available. Run 'evergreen get-update --install' to download and install it.
test/evergreen.yml is valid
=-=-=-=-=-=-=-=-=-=-=
Validating evergreen_develop.yml
A new version is available. Run 'evergreen get-update --install' to download and install it.
ERROR: unmarshalling project config from YAML: invalid configuration file: yaml: line 29: did not find expected key
ERROR: load project error(s): error unmarshalling yaml strict: yaml: line 29: did not find expected key
test/evergreen_develop.yml is an invalid configuration
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
```

File: dist/s_all [hunk 5]
```diff
@@ -91,33 +91,23 @@ run "python log.py"
...
+run "sh ./s_style $fast"
+run "./s_clang_format $fast"
 run "python prototypes.py"
 run "sh ./s_typedef -b"
 run "python test_tag.py"
 # The s_mentions script requires bash.
 run "./s_mentions" "--warning-only"
-if [ $(command -v evergreen) ] && [ -f ~/.evergreen.yml ]; then
-    run "cd .."
-    run "evergreen validate -p wiredtiger test/evergreen.yml"
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply dist/s_all:108]: f80b73bbc

File: dist/s_evergreen_validate [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    fi
+    cd ..
+    evergreen validate -p wiredtiger test/evergreen.yml > dist/$t 2>&1
+    exit1=$?
+    evergreen validate -p wiredtiger test/evergreen_develop.yml >> dist/$t 2>&1
+    exit2=$?
+    cd dist
+fi
+
+if test $exit1 -eq 1 -o $exit2 -eq 1 ; then
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply dist/s_evergreen_validate]: f80b73bbc

File: dist/s_all [hunk 7]
```diff
@@ -91,33 +91,23 @@ run "python log.py"
...
 COMMANDS="
 2>&1 ./s_define > ${t_pfx}s_define
 2>&1 ./s_docs > ${t_pfx}s_docs
 2>&1 ./s_evergreen > ${t_pfx}s_evergreen
+2>&1 ./s_evergreen_validate $fast > ${t_pfx}s_evergreen_validate
 2>&1 ./s_export > ${t_pfx}s_export
 2>&1 ./s_free > ${t_pfx}s_free
 2>&1 ./s_funcs > ${t_pfx}s_funcs
-2>&1 ./s_function > ${t_pfx}s_function
+2>&1 ./s_function $fast > ${t_pfx}s_function
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply dist/s_all]: f80b73bbc

**Review by lukech — APPROVED** [review]
LGTM

File: dist/s_evergreen_validate [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    *)
+        if test $# -gt 1 ; then
+            echo "Usage: $0 [-F]"
+            echo "-F only run validation commands if evergreen yml files have been modified"
+        fi
+        break;;
+esac
+
+if [ $(command -v evergreen) ] && [ -f ~/.evergreen.yml ]; then
+    if test $fast -eq 0 ; then
```
Comment (ddanderson) [inline dist/s_evergreen_validate:20]: Nit: I think code like this is clearer if you set `fast=true` or `fast=false`, and then this becomes:
```
  if $fast; then
```
(to get the negation you can do `if ! $fast; then`).

**Review by ddanderson — APPROVED** [review]
LGTM!  Just a nit to consider.

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Pull request merged**